crates/sovereign-sdk/full-node/sov-prover-storage-manager/src/snapshot_manager.rs

```diff
@@ -119,6 +119,10 @@ impl SnapshotManager {
 
         Ok(SnapshotManagerIter::new(db_iter, snapshot_iterators))
     }
+
+    pub fn get_db_handle(&self) -> Arc<sov_schema_db::DB> {
+        self.db.clone()
```

Comment:
It would be nice to see the consequences of returning a Weak ptr here. Just curious